### PR TITLE
Feature/3157 token detection v2

### PR DESCRIPTION
**Pull request by Cal-L**
**Description**

This is the implementation for token detection v2 as designed in Figma - https://www.figma.com/file/2h15YfvbT2q4q1FOMJFPP1/Token-Detection-V2?node-id=0%3A1. This is dependent on a controller PR - https://github.com/MetaMask/controllers/pull/763. In this iteration, we introduce a concept of detected tokens, where users will have a choice to either import or ignore certain tokens that has been "detected" in their wallet.

**Acceptance Criteria**

- WIP

**Issue**
#3157

**Figma**
- https://www.figma.com/file/2h15YfvbT2q4q1FOMJFPP1/Token-Detection-V2?node-id=0%3A1

### CODE REVIEW

File: app/components/UI/WhatsNewModal/index.js [hunk 1]
```diff
@@ -19,37 +19,62 @@ import { whatsNew } from './whatsNewList';
 import AsyncStorage from '@react-native-community/async-storage';
 import { CURRENT_APP_VERSION, LAST_APP_VERSION, WHATS_NEW_APP_VERSION_SEEN } from '../../../constants/storage';
 import compareVersions from 'compare-versions';
-import scaling from '../../../util/scaling';
 import PropTypes from 'prop-types';
 import { findRouteNameFromNavigatorState } from '../../../util/general';
+import StyledButton from '../StyledButton';
 import { useAppThemeFromContext, mockTheme } from '../../../util/theme';
+const modalMargin = 24;
+const modalPadding = 24;
```
Comment (gantunesr) [inline app/components/UI/WhatsNewModal/index.js:27]:
```suggestion
import StyledButton from '../StyledButton';
import { useAppThemeFromContext, mockTheme } from '../../../util/theme';

const modalMargin = 24;
const modalPadding = 24;
```

File: app/components/UI/WhatsNewModal/index.js [hunk 2]
```diff
@@ -19,37 +19,62 @@ import { whatsNew } from './whatsNewList';
...
 		},
-		button: {
+		slideContent: {
+			maxHeight: maxSlideItemHeight,
+		},
+		slideItemContainer: {
+			flex: 1,
+			width: slideItemWidth,
+			paddingHorizontal: modalPadding,
+			paddingBottom: 16,
```
Comment (gantunesr) [inline app/components/UI/WhatsNewModal/index.js:50]: Why some of this values are used in variables?

File: app/components/Views/AddAsset/index.js [hunk 3]
```diff
@@ -135,8 +136,8 @@ class AddAsset extends PureComponent {
 					</View>
 				)}
 				{assetType === 'token' ? (
-					<ScrollableTabView renderTabBar={() => this.renderTabBar()}>
-						{NetworksChainId.mainnet === this.props.chainId && (
+					<ScrollableTabView key={chainId} renderTabBar={() => this.renderTabBar()}>
```
Comment (gantunesr) [inline app/components/Views/AddAsset/index.js:139]: Why are you passing this prop? I don't see it the documentation

https://github.com/ptomasroos/react-native-scrollable-tab-view

File: ios/MetaMask.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -1008,6 +1008,7 @@
 				COPY_PHASE_STRIP = YES;
 				ENABLE_NS_ASSERTIONS = NO;
 				ENABLE_STRICT_OBJC_MSGSEND = YES;
+				"EXCLUDED_ARCHS[sdk=iphonesimulator*]" = arm64;
```
Comment (gantunesr) [inline ios/MetaMask.xcodeproj/project.pbxproj:1011]: Was this on purpose?

File: locales/languages/en.json [hunk 5]
```diff
@@ -1723,9 +1772,14 @@
...
-		"feature_security_settings_button": "View in Settings"
+		"feature_token_detection_title": "Improved token detection is here",
+		"feature_token_detection_description": "Improved token detection is currently available on Ethereum Mainnet, Polygon, BSC, and Avalanche network. More to come!\n\nWe source tokens from 3rd party tokens lists. Tokens listed on more than two token lists will be automatically detected.\n\nToken detection feature is ON by default. But you can disable it from Settings.",
+		"feature_token_detection_button_text": "Visit in Settings",
+		"feature_token_security_title_1": "Scam and security risks",
+		"feature_token_security_description_1": "Tokens can be created by anyone and can have duplicate names. If you see a token appear that you don’t trust or haven’t interacted with - its safer to not trust it.",
+		"feature_token_security_title_2": "Long press to remove tokens",
+		"feature_token_security_description_2": "On MetaMask mobile app, you can always remove tokens by long pressing on the token, and select ‘Remove’.",
```
Comment (gantunesr) [inline locales/languages/en.json:1781]:
```suggestion
		"feature_token_security_description_2": "On MetaMask mobile app, you can always remove tokens by long pressing on the token, and select ‘Hide’.",
```

**Review by gantunesr — COMMENTED** [review]
I went over most of the files today, in the review we can look with more detail the following files:

- `app/components/Views/AssetDetails/index.tsx`
- `app/components/Views/AssetOptions/index.tsx`
- `app/components/Views/DetectedTokens/components/Token.tsx`
- `app/components/Views/DetectedTokens/index.tsx`
- `app/components/Nav/Main/MainNavigator.js`(again)

File: app/constants/on-ramp.js [hunk 6]
```diff
@@ -49,6 +49,15 @@ export const NETWORKS_CHAIN_ID = {
 	FANTOM: '250',
 };
 
+export const FORMATTED_NETWORK_NAMES = {
+	[NETWORKS_CHAIN_ID.MAINNET]: 'Ethereum Mainnet',
+	[NETWORKS_CHAIN_ID.BSC]: 'Binance Smart Chain',
+	[NETWORKS_CHAIN_ID.POLYGON]: 'Polygon',
+	[NETWORKS_CHAIN_ID.AVAXCCHAIN]: 'Avalanche',
```
[the remark this answers is not in the document]
Comment (Cal-L) [reply app/constants/on-ramp.js:56]: Will handle in separate PR

File: ios/MetaMask.xcodeproj/project.pbxproj (hunk 4 above)
Comment (Cal-L) [reply ios/MetaMask.xcodeproj/project.pbxproj:1011]: Verify if this was removed at some point

**Pull request closed without merging**